sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasExec.scala

```diff
@@ -106,6 +107,37 @@ case class TransformWithStateInPandasExec(
     List.empty
   }
 
+  override def shouldRunAnotherBatch(newInputWatermark: Long): Boolean = {
+    if (timeMode == ProcessingTime) {
+      // TODO: check if we can return true only if actual timers are registered, or there is
```

Comment:
Since this PR is fairly light-weighted, should we include that check in the same PR? If not, please create a Jira to track the TODO and link it here.

Reply:
This is also the way it is on Scala side [here](https://github.com/apache/spark/blob/738dfa346bde59d10fa18bc662b858b0c3354254/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala#L88). Let me check if there is any specific reason for Bhuwan to leave this as a TODO here.

Reply:
Confirmed with Anish - the way we have it today is that we'll keep constructing new batches because we may have future timer expiring. I sent out a SPARK JIRA to keep track of this issue: https://issues.apache.org/jira/browse/SPARK-50180 and will update the comments in both Scala and Python.

Reply:
> Since this PR is fairly light-weighted, should we include that check in the same PR?

Some additional testing on AvailableNow trigger mode may be needed together with fixing the issue and it exceeds the description of this PR. I'll leave it as it is today and let's keep track of the fix in the [SPARK-50180](https://issues.apache.org/jira/browse/SPARK-50180)